src/rgw/rgw_user.cc

```diff
@@ -1735,6 +1735,7 @@ int RGWUser::init(RGWUserAdminOpState& op_state)
   }
 
   if (!uid.empty() && (uid.compare(RGW_USER_ANON_ID) != 0)) {
+    op_state.set_reserved_user(false);
```

Comment:
> Then the reserved_user should be false right ?

Yes

>  I understand that anonymous user and reserved_user are different.

Yes, anonymous user is one kind of reserved user. Future more maybe we can make server group user as reserved too.

The `reserved_user` becomes true in `RGWUserAdminOpState` initialization phrase, see
https://github.com/ceph/ceph/pull/16005/files#diff-5ef7b455247dea8b72d475e1ce348284R494